scipy/signal/tests/test_bsplines.py

```diff
@@ -255,6 +255,33 @@ def test_sepfir2d_invalid_image():
     with pytest.raises(ValueError, match="object of too small depth"):
         signal.sepfir2d(image[0], filt, filt)
 
+def test_sepfir2d():
+    np.random.seed(181819140)
+    filt = np.array([1.0, 2.0, 4.0, 2.0, 1.0, 3.0, 2.0])
+    image = np.random.rand(4, 4)
+
+    expected = array([[33.772956, 31.264141, 36.075042, 39.040313],
+       [38.942419, 36.574301, 37.060368, 37.403998],
+       [37.563456, 35.758501, 33.46929 , 30.447159],
+       [24.466602, 23.506581, 24.316481, 19.806113]])
+    assert_allclose(signal.sepfir2d(image, filt, filt[::3]), expected)
```

Comment:
The MacOS CI failure looks related at least:

```
=================================== FAILURES ===================================
________________________________ test_sepfir2d _________________________________
scipy/signal/tests/test_bsplines.py:267: in test_sepfir2d
    assert_allclose(signal.sepfir2d(image, filt, filt[::3]), expected)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=0
E   
E   Mismatched elements: 2 / 16 (12.5%)
E   Max absolute difference: 31.99999962
E   Max relative difference: 1.02353683
```

Reply:
Do I need to do something else for MacOS. Locally, running the tests for `signal` submodule on Linux passes for me:
```py
scipy/signal/tests/test_bsplines.py::test_sepfir2d PASSED                [  0%]
...
============== 2787 passed, 2 skipped, 10475 deselected in 53.99s ==============

```

Reply:
This passes for me on macOS 11 with the conda compilers. The issue may be specific to GCC.